pkg/dynamic/field.go

```diff
@@ -24,3 +27,39 @@ func LookupSymbolField(rs reflect.Value) (string, bool) {
 	return field.String(), true
 }
 
+// Used by bbgo/interact_modify.go
+func GetModifiableFields(val reflect.Value, callback func(tagName, name string)) {
+	for i := 0; i < val.Type().NumField(); i++ {
+		t := val.Type().Field(i)
+		if !t.IsExported() {
+			continue
+		}
+		modifiable := t.Tag.Get("modifiable")
+		if modifiable != "true" {
+			continue
+		}
+		jsonTag := t.Tag.Get("json")
+		if jsonTag == "" || jsonTag == "-" {
+			continue
+		}
+		name := strings.Split(jsonTag, ",")[0]
+		callback(name, t.Name)
+	}
+}
+
+var ZeroValue reflect.Value = reflect.Zero(reflect.TypeOf(0))
+
+func GetModifiableField(val reflect.Value, name string) (reflect.Value, bool) {
+	field, ok := val.Type().FieldByName(name)
+	if !ok {
+		return ZeroValue, ok
```

Comment:
it's better to create the zero reflect value from it's field type instead of `0`

Reply:
non-existing field will not have type. any error will be presented in ok (the second return value)